### PR TITLE
Fix issues

### PR DESCRIPTION
**Pull request by AnjanaSenanayake**
Fix issues

- https://github.com/wso2-enterprise/open-healthcare/issues/1477
- https://github.com/wso2-enterprise/open-healthcare/issues/1478
- https://github.com/wso2-enterprise/open-healthcare/issues/1495

### CODE REVIEW

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java [hunk 1]
```diff
@@ -71,11 +71,13 @@ public Map<String, DatatypeTemplateContext> getDataTypeTemplateContextMap() {
     private void populateDatatypeContext() {
         for (Map.Entry<String, FHIRDataTypeDef> datatypeDefnEntry : datatypeDefnMap.entrySet()) {
             FHIRDataTypeDef datatypeDefn = datatypeDefnEntry.getValue();
-            if (DEFAULT_DATA_TYPES.contains(datatypeDefn.getDefinition().getName())) {
+            if (DEFAULT_DATA_TYPES.contains(datatypeDefn.getDefinition().getName()) || "Extension".equals(datatypeDefn.getDefinition().getType())) {
```
Comment (sameeragunarathne) [inline native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java:74]: Formatting issue?

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java [hunk 2]
```diff
@@ -71,11 +71,13 @@ public Map<String, DatatypeTemplateContext> getDataTypeTemplateContextMap() {
...
                 continue;
             }
             DatatypeTemplateContext context = new DatatypeTemplateContext();
-            context.setName(GeneratorUtils.getInstance().getUniqueIdentifierFromId(datatypeDefn.getDefinition().getId()));
+            String typeName = CommonUtil.getSplitTokenAt(datatypeDefn.getDefinition().getUrl(), "/", ToolConstants.TokenPosition.END);
+            typeName = GeneratorUtils.getInstance().getUniqueIdentifierFromId(typeName);
+            context.setName(GeneratorUtils.getInstance().getUniqueIdentifierFromId(typeName));
```
Comment (ThishaniLucas) [inline native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java:80]: typeName going through GeneratorUtils.getInstance().getUniqueIdentifierFromId() twice?

Comment (AnjanaSenanayake) [reply native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/2cbe55c59fcc81e82a4423d75f58276b0d6374a4

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java (hunk 1 above)
Comment (AnjanaSenanayake) [reply native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/DatatypeContextGenerator.java]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/3094d8dcd8da09b17744e25235482c20e2611f1a

**Review by isuruh15 — APPROVED** [review]

**Review by sameeragunarathne — APPROVED** [review]

**Review by nirmal070125 — APPROVED** [review]

**Pull request merged**